pwa/src/templates/mappingDetailTemplate/MappingDetailsTemplate.tsx

```diff
@@ -121,19 +127,24 @@ export const MappingDetailTemplate: React.FC<MappingDetailsTemplateProps> = ({ m
                   <div className={styles.content}>
                     <FormField>
                       <FormFieldInput>
-                        <div className={styles.formFieldInfoHeader}>
-                          <FormFieldLabel>{t("Input")}</FormFieldLabel>
-                          <ToolTip tooltip="The input is the JSON code you want to test the mapping with.">
-                            <FontAwesomeIcon icon={faInfoCircle} />
-                          </ToolTip>
+                        <div className={styles.formFieldHeader}>
+                          <div className={styles.formFieldInfoHeader}>
+                            <FormFieldLabel>{t("Input")}</FormFieldLabel>
+                            <ToolTip tooltip="The input is the JSON code you want to test the mapping with.">
+                              <FontAwesomeIcon icon={faInfoCircle} />
+                            </ToolTip>
+                          </div>
+
+                          <FormFieldLabel>
+                            <a onClick={beautifyEditorCode} className={styles.beautifyButton}>
+                              <FontAwesomeIcon icon={faMagicWandSparkles} />
+                              Beautify
+                            </a>
+                          </FormFieldLabel>
                         </div>
 
-                        <Textarea
-                          {...{ register, errors }}
-                          name="input"
-                          validation={{ validate: validateStringAsJSON }}
-                          disabled={isLoading.mappingForm}
-                        />
+                        <CodeEditor {...{ beautify, editorRef }} />
```

Comment:
I think we should rethink the idea of this being a form. I do agree that it has some overlap, the code is just an input that needs to be sent.

However, now we're also adding the Beautify button into a `FormFieldLabel`, which I don't think makes a lot of sense.

I'd probably opt to keep it outside of a form, and write some custom logic.

e.g. (dummy code; based on other comments as well).


```
const { Editor, code, setCode } = useCodeEditor("optional initial code value");

...

<>
  <Button disabled={!code} onClick={() => alert("testing!")}>Test mapping</Button>
  
  <Editor beautify /> // the beautify logic can live within the useCodeEditor hook, unsure if it even needs to be optional, might be fine to just _always_ show it (if so, then the implementation would simply become <Editor />
</>
```

This way, the `Editor` is much easier to be re-used. Also: I think in the current implementation, `setCode` might not even be required.